python/lsst/ap/association/dia_collection.py

```diff
@@ -168,18 +197,38 @@ def score(self, dia_source_catalog, max_dist_arcsec):
         dia_source_catalog : an lsst.afw.SourceCatalog
             A contiguous catalog of dia_sources to "score" based on distance
             and (in the future) other metrics.
-        max_dist_arcsec : float
+        max_dist : lsst.afw.geom.Angle
             Maximum allowed distance to compute a score for a given DIAObject
             DIASource pair.
 
         Returns
         -------
-        (N DIAObject by M DIASource) float array of between DIAObjects and
-        DIASources.
+        lsst.pipe.base.Struct
+            struct containing:
+            * scores: array of floats of match quality
+            * indices: index in DIAObjectCollection that source matched to
+            Default values for these arrays are NaN and the number of
+            DIAObjects in this collection, respectively.
         """
-        pass
+        if not self._is_valid_tree:
+            self.update_spatial_tree()
+
+        scores = np.ones(len(dia_source_catalog)) * np.nan
```

Comment:
If you're just making an array of NaNs, use `np.empty`, since you don't need to initialize it.

Reply:
I need the the NaNs to have a positive sign so that they end up at the end of a sort. I'd like to keep this then, unless there is a way to do this with np.empty

Reply:
Per our conversation, NaN isn't signed, and what you really want is `np.inf`(or the negative thereof).